apps/explorer/src/app/hooks/use-txs-data.ts

```diff
@@ -65,6 +65,14 @@ export const useTxsData = ({ limit, filters }: IUseTxsData) => {
     }
   }, [setTxsState, data]);
 
+  useEffect(() => {
+    setTxsState((prev) => ({
```

Comment:
setting state from an effect is an anti pattern. Can this state setter be lifted?

Reply:
You are right and in all normal scenarios I should rewrite this. However the next ticket is junking this loader and rewriting it entirely (#4070) so instead I will make sure I don't port the mistake there.